examples/algebra/ring/ringDividesScript.sml

```diff
@@ -29,7 +29,10 @@ open ringIdealTheory;
 open ringUnitTheory;
 open ringTheory;
 open groupTheory;
-open monoidTheory;
+open monoidTheory gbagTheory bagTheory;
+val MEMBER_NOT_EMPTY = pred_setTheory.MEMBER_NOT_EMPTY;
```

Comment:
There's an `open pred_setTheory` earlier in this file; do you need this rebind of `MEMBER_NOT_EMPTY` because the name is masked by some other `open`?  If so, perhaps we should avoid the name clash in the relevant theories.

Reply:
It's masked by `bagTheory.MEMBER_NOT_EMPTY`. Yes perhaps we should avoid the name clash but could be done separately from this PR (and not by me :))?

Reply:
Totally.  I can change the one in `bagTheory`.